### ISSUE TITLE
Uncaught TypeError: Argument 1 passed to MultiSafepay\\ConnectCore\\Logger\\Logger::logPaymentLinkError() must be of the type string, null given

### ISSUE THREAD

**Issue by hostep** [MultiSafepay/magento2#35]
**Describe the bug**
Using:
- Magento OS 2.3.6-p1
- Following MultiSafepay modules:
  - multisafepay/magento2 version 2.19.0
  - multisafepay/magento2-adminhtml version 1.19.0
  - multisafepay/magento2-catalog-inventory version 1.0.2
  - multisafepay/magento2-core version 2.17.0
  - multisafepay/magento2-frontend version 1.16.1
  - multisafepay/magento2-msi version 1.0.1
  - multisafepay/php-sdk version 5.6.0

I've found the following error report in `var/report/api/{some-number}`:

```
"Fatal Error: 'Uncaught TypeError: Argument 1 passed to MultiSafepay\\ConnectCore\\Logger\\Logger::logPaymentLinkError() must be of the type string, null given, called in \/vendor\/multisafepay\/magento2-core\/Gateway\/Http\/Client\/RecurringClient.php on line 72 and defined in \/vendor\/multisafepay\/magento2-core\/Logger\/Logger.php:37
Stack trace:
#0 \/vendor\/multisafepay\/magento2-core\/Gateway\/Http\/Client\/RecurringClient.php(72): MultiSafepay\\ConnectCore\\Logger\\Logger->logPaymentLinkError(NULL, Object(MultiSafepay\\Exception\\ApiException))
#1 \/vendor\/magento\/module-payment\/Gateway\/Command\/GatewayCommand.php(106): MultiSafepay\\ConnectCore\\Gateway\\Http\\Client\\RecurringClient->placeRequest(Object(Magento\\Payment\\Gateway\\Http\\Transfer))
#2 \/vendor\/magento\/module-payment\/Gateway\/Command\/CommandManager.php(65): Magent' in '\/vendor\/multisafepay\/magento2-core\/Logger\/Logger.php' on line 37
```

Is there a known case where `$orderId` could be `null` [over here](https://github.com/MultiSafepay/magento2-core/blob/2.17.0/Gateway/Http/Client/RecurringClient.php#L72)? If there is a chance it can be `null`, that code will probably have to be rewritten a little bit to be able to deal with that?

**To Reproduce**
1. No idea, found this in `var/report/api/{some-number}`

**Expected behavior**
No errors

**Screenshots**
N/A

**Desktop (please complete the following information):**
 - OS: [e.g. iOS]: Not relevant
 - Browser [e.g. chrome, safari]: Not relevant
 - Version [e.g. 22]: Not relevant

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6]: Not relevant
 - OS: [e.g. iOS8.1]: Not relevant
 - Browser [e.g. stock browser, safari]: Not relevant
 - Version [e.g. 22]: Not relevant

**Additional context**
N/A

**Comment by hostep** [comment]
The `multisafepay.log` contains (next to a bunch of info & debug messages) one error in the same minute where that report file was created:

```
[2022-08-22 06:44:14] MultiSafepay.ERROR: The MultiSafepay API key is invalid. Invalid API key [] []
```
(UTC timestamp)

Which is a weird error since there are plenty of transactions that work fine and all using the same API key ...
In fact, I count 308 such errors today only on this shop, which is really unexpected...

**Comment by vinodsowdagar** [comment]
Hi @hostep ,

We've improved the logging so at least the error is not going to show up anymore whenever the Order ID is not present. What exactly is happening related to the retrieval of the payment link itself we could unfortunately not yet figure out. 
More information about this will follow.

**Comment by hostep** [comment]
As far as I can see, this was fixed in [version 2.20.1](https://github.com/MultiSafepay/magento2/releases/tag/2.20.1) of this module, so let's close this ticket. Thanks!

If I'm mistaken, let me know 🙂
